app/services/CheckoutService.scala

```diff
@@ -307,17 +306,22 @@ class CheckoutService(
 
   private def createPaymentType(purchaserIds: PurchaserIdentifiers, subscriptionData: SubscribeRequest): Future[NonEmptyList[SubsError] \/ PaymentService#Payment] = {
 
+    val deliveryInstructions: Option[String] = subscriptionData.productData match {
+      case Left(paperData) => paperData.deliveryInstructions
+      case _ => None
+    }
+
     try {
       val payment = subscriptionData.genericData.paymentData match {
         case paymentData@DirectDebitData(_, _, _) =>
           val personalData = subscriptionData.genericData.personalData
           require(personalData.address.country.contains(Country.UK), "Direct Debit payment only works in the UK right now")
-          paymentService.makeDirectDebitPayment(paymentData, personalData.first, personalData.last, purchaserIds)
+          paymentService.makeDirectDebitPayment(paymentData, personalData.first, personalData.last, purchaserIds, deliveryInstructions)
         case paymentData@CreditCardData(_) =>
           val plan = subscriptionData.productData.fold(_.plan, _.plan)
           val desiredCurrency = subscriptionData.genericData.currency
           val currency = if (plan.charges.price.currencies.contains(desiredCurrency)) desiredCurrency else GBP
-          paymentService.makeCreditCardPayment(paymentData, currency, purchaserIds)
+          paymentService.makeCreditCardPayment(paymentData, currency, purchaserIds, deliveryInstructions)
```

Comment:
I think the biggest confusion is that Payment is defined here as :
  sealed trait Payment {
    def makeAccount: Account
    def makePaymentMethod: Future[PaymentMethod]
  }

which to me looks more like a payment method/ account builder